bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -804,11 +809,23 @@ private boolean updateChannels(Resource resource) {
             case BUTTON:
                 if (fullUpdate) {
                     addSupportedChannel(CHANNEL_2_BUTTON_LAST_EVENT);
+                    addSupportedChannel(CHANNEL_2_BUTTON_LAST_UPDATED);
                     controlIds.put(resource.getId(), resource.getControlId());
                 } else {
                     State buttonState = resource.getButtonEventState(controlIds);
                     updateState(CHANNEL_2_BUTTON_LAST_EVENT, buttonState, fullUpdate);
                 }
+                // Update channel from timestamp if last button pressed.
+                State buttonLastUpdatedState = resource.getButtonLastUpdatedState(timeZoneProvider.getTimeZone());
+                if (buttonLastUpdatedState instanceof DateTimeType) {
+                    Instant buttonLastUpdatedInstant = ((DateTimeType) buttonLastUpdatedState).getInstant();
+                    if (buttonLastUpdatedInstant.isAfter(lastUpdated)) {
+                        updateState(CHANNEL_2_BUTTON_LAST_UPDATED, buttonLastUpdatedState, fullUpdate);
+                        lastUpdated = buttonLastUpdatedInstant;
+                    }
+                } else if (Instant.MIN.equals(lastUpdated)) {
+                    updateState(CHANNEL_2_BUTTON_LAST_UPDATED, buttonLastUpdatedState, fullUpdate);
+                }
```

Comment:
Note: This "tricky" (messy) logic is because we have one DTO per button, and this is modelled into one Thing and one corresponding channel. Since the API actually provides a timestamp per button, we are only interested in the last timestamp (i.e. when "a button" was last pressed).

This can cause multiple updates, because we are receiving the responses unordered. However, the scope was already expanded more than expected for this PR. It's a bit of a work-around, and I think we have to come back to this. To fix this I think we need to collect the button responses and only process them once we have received all. For initial state this seems doable. For push messages we probably receive only messages for buttons actually pushed, so it should be fine already.

Reply:
> collect the button responses and only process them once we have received all.

That is probably a bad idea. If the user presses and holds then the press event must be processed immediately, and the release event pricessed later.

Reply:
I only suggest this when retrieving initial state, for example when requesting and processing all data during initialization:
> For initial state this seems doable.

When receiving live data, I agree we shouldn't do this:
> For push messages we probably receive only messages for buttons actually pushed, so it should be fine already.

Reply:
For initial state you can probably do (pseudo code)..

```
if (newLastUpdatedInstant.isAfter(currentChannelLastUpdatedInstantValue)) {
  updateChannel(newLastUpdatedInstant);
}

Reply:
I believe it's already like that? The only problem that remains for now is that it can cause multiple updates.

Reply:
> I believe it's already like that?

So it is. But I did request that you change the `lastUpdate` field to `buttonGroupLastUpdated` for clarity.